### DISCUSSION TITLE
오픈 프로필 id와 실제 프로필 id

### DISCUSSION THREAD

**Discussion by gd1422333** [storycraft/node-kakao#652] [Q&A] [answered]
몇 주 전부터 오픈 프로필 id와 실제 프로필 id의 로직을 찾는것에 큰 흥미를 느껴 로직을 찾는 시도를 수 차례 해보았습니다. 그 결과 2020년 8월 이전에 생성된 채팅방에서는 오픈 프로필 id와 실제 프로필 id의 로직이 확실히 존재한다는 것을 알았고 이 로직을 똑같이 2021년 이후에 생성된 채팅방에서 적용 해보았으나 전혀 구해지지 않았습니다. 로직을 수 차례 수정해서 적용 해보아도 실제 프로필의 id를 구하기에는 해당 로직이 무용지물이었습니다. 궁금증을 가지고 2021년 이후 생성된 채팅방과 2020년 8월 이전에 생성된 채팅방의 오픈 프로필 id를 비교 해보았는데 2020년 8월 이전 채팅방에는 해당 방에 소속되있는 유저들의 오픈 프로필 id 초반 부분이 다 같았고 2021년 이후는 오픈 프로필 id 자체가 각기 달랐습니다. 또 2020년 8월 이전의 오픈 프로필 id보다 최신 2021년 오픈채팅방의 id가 2자리 정도 많다는 것도 알게 되었습니다. 이러한 사실들을 가지고 끊임없이 시도를 해보았는데 결과적으로 얻은 건 아무것도 없었습니다. 모든 경우의 수를 동원하여 2020년 8월 이전 채팅방의 실제 프로필 id를 얻는 로직을 전부 수정해 보아 사용했지만 도저히 답을 찾을 수 없었습니다. 해당 채팅방의 linkId와 관련이 있을까도 생각하여 여러 시도를 또 다시 해보았지만 결과는 여전했습니다. 결국 뻘짓만 하다 로직이 존재하지 않는다 라는 결론을 내리고 로직 찾는 연구를 그만뒀었는데 최근에 어떠한 주식방에서 해외계정들을 전부 내보내는 챗봇을 운영한다는 사실을 알게 되었습니다. 입장하여 보니 해외계정들이 정말로 입장과 동시에 바로 내보내지는것을 확인하였습니다. 이 토론에서도 2021년 이후 실제 프로필 id를 구하는 로직을 알고 계시는 분들이 있다고 들었는데, 카카오톡 자체를 후킹해야 로직을 구할 수가 있는건지 여부가 궁금합니다. 더 많은 시도도 안 해보고 무책임하게 방법만 얻으려 하는 것이 아니라 그냥 단순히 카카오톡을 후킹해야 해당 로직을 획득할 수 있는지에 대한 여부가 궁금할 뿐입니다. 답변 꼭 부탁드리겠습니다.

**Comment by wiresharkkkk** [comment]
흥미로운 주제네요 눈은 좀 아프지만..

**Comment by benesiaxd** [comment, marked as answer]
미치고 팔짝뛰겠네요 ㅋㅋ
어떻게하면 불법을 행할수있을지 당당하게 질문하는 모습이 멋지십니다.

**Reply by gd1422333** [reply]
정확한 기준도 없이 그냥 자기 줏대로 합법이니 불법이니 판단하시는데 참 미치고 팔짝 뛰겠어요~ㅋㅋ

**Reply by gd1422333** [reply]
답변 드릴 생각 없으시면 괜한 오지랖 부리지 말고 갈 길이나 가주시면 됩니다. 급식같다 뭐같다라고 표현하시는 님이 오히려 더 급식인것 같은데요ㅋㅋㅋ 성인이시면 저런 어린애 글 하나에 못 참아서 정의의 사도마냥 부들부들 지적질할 필요가 없으시지 않나요? 참으로 나잇값을 잘 하시네요. 미치고 팔짝 뛰겠습니다~

**Comment by ghost** [comment]
후킹과 실제 카카오톡 사용자의 유저 ID랑 무슨 관련이 있겠습니까.
